app/models/middleware_server.rb

```diff
@@ -33,6 +37,21 @@ def evaluate_alert(alert_id, event)
   end
 
   def in_domain?
-    !middleware_server_group.nil?
+    middleware_server_group
+  end
+
+  def hawkular?
+    product == 'Hawkular'
+  end
+
+  def immutable?
+    hawkular? ||
+      ['Immutable', 'Is Immutable'].any? do |key|
```

Comment:
please add it as a comment in code for easier understanding by anyone who reads it